### DISCUSSION TITLE
How to generate a variable that only contain the "name" of legislators who mentioned "health insurance"?

### DISCUSSION THREAD

**Discussion by joufeihuang** [davidycliao/legisTaiwan#1] [Q&A] [answered]
I want to generate a variable that identifies "legislators' names" who have mentioned "health insurance" from the get_parlquestions(). Does anyone know how to do it? There is a pattern in the "item" from the get_parlquestions(). 
"本院.委員..", so I thought I could use str_extract and regular expression to extract this "name" variable. However, the following code is not working. Any suggestion

`question_8 <- get_parlquestions(term = 8)
onlydf_not_onlyfan <- question_8$data

df <- onlydf_not_onlyfan[str_detect(onlydf_not_onlyfan[["item"]], "健康保險", negate = FALSE),]

df<- as.data.frame(df)

str_extract(regexpr("本院", df$item), df$item)
`

**Comment by davidycliao** [comment]
library(legisTaiwan)
library(purrr)
library(stringr)

question_8 <- get_parlquestions(term = 8)
onlydf_not_onlyfan <- question_8$data
df <- onlydf_not_onlyfan[str_detect(onlydf_not_onlyfan[["item"]], "健康保險", negate = FALSE),]


x <- df$item |>
  map(function(x){strsplit(x, "、|，")[[1]][2]}) |>
  map_chr(function(x){str_remove_all(x, paste(c("本院", "委員"), collapse = "|"))})

df["legislators"] <- x
df[c("legislators", "item")]
> df[c("legislators", "item")]
# A tibble: 14 × 2
   legislators item                                                                                                 
   <chr>       <chr>                                                                                                
 1 蔡煌瑯      二八六、本院蔡委員煌瑯，針對美國在台協會主席薄瑞光先生拜會馬總統之後，農委會可能開立大門，設下「一國…
 2 盧秀燕      二九五、本院盧委員秀燕，針對我國目前對於懷孕婦女之照顧，僅於中央健康保險局的規劃中，提供產前檢查之給…
 3 黃偉哲      三八五、本院黃委員偉哲，鑒於縣市升格後行政院擬修正財政收支劃分法，將勞工保險費及全民健康保險費全數由…
 4 徐欣瑩      六十七、本院徐委員欣瑩，有鑑於二代健保乃是為解決健保財務問題，同時處理一代健保對一般受薪家庭保費占家…
 5 盧秀燕      七十四、本院盧委員秀燕，針對全民健康保險施行細則係針對全民健康保險法所補充不足而定細項規則，未經過立…
 6 廖國棟      四十三、本院廖委員國棟，為前衛生署公告之「全民健康險藥品價格調整作業辦法」有逾越母法授權之虞，因該辦…
 7 葉津鈴      十八、本院葉委員津鈴，就103年1月1日起全民健康保險調高第二類、第三類被保險人的最低投保薪資級距，惟受… 
 8 楊瓊瓔      八十六、本院楊委員瓊瓔，針對現行「全民健康保險法」規定健保費按月計算，並非按投保實際日計算，導致新生…
 9 江惠貞      八十九、本院江委員惠貞，鑒於衛福部於103年3月20日公告《全民健康保險藥品價格調整作業辦法》中「將逾（無…
10 黃昭順      一二○、本院黃委員昭順，針對台灣社會面臨的人口老化挑戰，再次提醒國人絕對不能掉以輕心，籲請政府更積極… 
11 邱志偉      三十、本院邱委員志偉，台灣高等教育產業工會於去年11月推動「學校欠我勞健保」集體檢舉行動，募集49名學生…
12 江惠貞      二十四、本院江委員惠貞，有鑑於我國全民健康保險自民國84年實施以來，以「風險連帶」、「同舟共濟」為主要…
13 邱志偉      四十七、本院邱委員志偉，鑑於日前報載，由於台灣高齡、少子化，全民健康保險將可能於三年後入不敷出，爰此…
14 林國正      十五、本院林委員國正，鑒於我國現行健保補充保費制度尚無法靈活因應社會現況，並為保障基層民眾以渡景氣嚴…

**Reply by joufeihuang** [reply]
Thank you so much!

**Reply by davidycliao** [reply, marked as answer]
😉
